cmd/nerdctl/main.go

```diff
@@ -134,44 +132,8 @@ func xmain() error {
 	return app.Execute()
 }
 
-// Config corresponds to nerdctl.toml .
-// See docs/config.md .
-type Config struct {
-	Debug            bool     `toml:"debug"`
-	DebugFull        bool     `toml:"debug_full"`
-	Address          string   `toml:"address"`
-	Namespace        string   `toml:"namespace"`
-	Snapshotter      string   `toml:"snapshotter"`
-	CNIPath          string   `toml:"cni_path"`
-	CNINetConfPath   string   `toml:"cni_netconfpath"`
-	DataRoot         string   `toml:"data_root"`
-	CgroupManager    string   `toml:"cgroup_manager"`
-	InsecureRegistry bool     `toml:"insecure_registry"`
-	HostsDir         []string `toml:"hosts_dir"`
-	Experimental     bool     `toml:"experimental"`
-}
-
-// NewConfig creates a default Config object statically,
-// without interpolating CLI flags, env vars, and toml.
-func NewConfig() *Config {
-	return &Config{
-		Debug:            false,
-		DebugFull:        false,
-		Address:          defaults.DefaultAddress,
-		Namespace:        namespaces.Default,
-		Snapshotter:      containerd.DefaultSnapshotter,
-		CNIPath:          ncdefaults.CNIPath(),
-		CNINetConfPath:   ncdefaults.CNINetConfPath(),
-		DataRoot:         ncdefaults.DataRoot(),
-		CgroupManager:    ncdefaults.CgroupManager(),
-		InsecureRegistry: false,
-		HostsDir:         ncdefaults.HostsDirs(),
-		Experimental:     true,
-	}
-}
-
 func initRootCmdFlags(rootCmd *cobra.Command, tomlPath string) (*pflag.FlagSet, error) {
-	cfg := NewConfig()
+	cfg := config.New()
```

Comment:
Is it better to init `GlobalCommandOptions` here and then pass it to `config.New()`
```
cfg := GlobalCommandOptions{...}
config.New(cfg)
```
to keep the code semantic.
Cc @Zheaoli